diff --git a/node_modules/smpp/lib/defs.js b/node_modules/smpp/lib/defs.js
index 0690488..d8442d4 100644
--- a/node_modules/smpp/lib/defs.js
+++ b/node_modules/smpp/lib/defs.js
@@ -370,7 +370,7 @@ filters.message = {
 		}
 		var message = typeof value == 'string' ? value : value.message;
 		if (typeof message == 'string') {
-			var encoding = encodings.detect(message);
+			var encoding = value.encoding||encodings.detect(message);
 			if (message && this.data_coding === null) {
 				this.data_coding = consts.ENCODING[encoding];
 			}
